google-cloud-spanner/src/main/java/com/google/cloud/spanner/TransactionRunnerImpl.java

```diff
@@ -447,7 +463,13 @@ TransactionSelector getTransactionSelector() {
             // Aborted error if the call that included the BeginTransaction option fails. The
             // Aborted error will cause the entire transaction to be retried, and the retry will use
             // a separate BeginTransaction RPC.
-            TransactionSelector.newBuilder().setId(tx.get()).build();
+            if (trackTransactionStarter) {
+              TransactionSelector.newBuilder()
+                  .setId(tx.get(waitForTransactionTimeoutMillis, TimeUnit.MILLISECONDS))
```

Comment:
Use a timeout if we are tracking the statement that should return a transaction id.